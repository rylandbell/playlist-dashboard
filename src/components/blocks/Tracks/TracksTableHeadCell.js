import React, { Component } from 'react';

class TracksTableHeadCell extends Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.props.handleColumnHeadingClick(this.props.feature.name);
  }

  render() {
    let sortIndicator;
    if (this.props.feature.sortBy === 'ascending') {
      sortIndicator = <span className="fa fa-sort-asc" />
    } else if (this.props.feature.sortBy === 'descending') {
      sortIndicator = <span className="fa fa-sort-desc" />
    } else {
      sortIndicator = <span className="transparent fa fa-sort-desc" />
    }

    return (
      <th className="tracks__table-head-cell text-center hidden-xs noselect" onClick={this.onClick}>
        {this.props.feature.shortName}&nbsp;
        {sortIndicator}
      </th>
    );
  }
}

export default TracksTableHeadCell;
