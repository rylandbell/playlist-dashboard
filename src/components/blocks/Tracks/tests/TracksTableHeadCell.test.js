import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import TracksTableHeadCell from "../TracksTableHeadCell";
import { testFeaturesData, testFilteredTracks } from "./test-data";

describe("TracksTableHeadCell component", () => {
  const props = {
  feature: testFeaturesData[0],
  handleColumnHeadingClick: () => {}
};

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TracksTableHeadCell {...props} />, div);
  });

  it("shallow render matches snapshot", () => {
    const tree = toJson(shallow(<TracksTableHeadCell {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
