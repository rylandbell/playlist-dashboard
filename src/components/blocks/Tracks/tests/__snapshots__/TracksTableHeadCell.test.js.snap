// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksTableHeadCell component shallow render matches snapshot 1`] = `
<th
  className="tracks__table-head-cell text-center hidden-xs noselect"
  onClick={[Function]}
>
  Dance
  Â 
  <span
    className="transparent fa fa-sort-desc"
  />
</th>
`;
